site/src/pages/WorkspacePage/WorkspaceNotifications.tsx

```diff
@@ -0,0 +1,343 @@
+      notifications.push({
+        title: "Autostart has been disabled for your workspace.",
+        severity: "warning",
+        detail:
+          "Autostart is unable to automatically update your workspace. Manually update your workspace to reenable Autostart.",
+
+        actions,
+      });
+    } else {
+      notifications.push({
+        title: "An update is available for your workspace",
+        severity: "info",
+        detail: latestVersion.message,
+        actions,
+      });
+    }
+  }
+
+  // Unhealthy
+  if (
+    workspace.latest_build.status === "running" &&
+    !workspace.health.healthy
+  ) {
+    notifications.push({
+      title: "Workspace is unhealthy",
+      severity: "warning",
+      detail: (
+        <>
+          Your workspace is running but{" "}
+          {workspace.health.failing_agents.length > 1
+            ? `${workspace.health.failing_agents.length} agents are unhealthy`
+            : `1 agent is unhealthy`}
+          .
+        </>
+      ),
+      actions: permissions.updateWorkspace ? (
+        <NotificationActionButton onClick={onRestartWorkspace}>
+          Restart
+        </NotificationActionButton>
+      ) : undefined,
+    });
+  }
+
+  // Dormant
+  const areActionsEnabled = useIsWorkspaceActionsEnabled();
+  if (areActionsEnabled && workspace.dormant_at) {
+    const formatDate = (dateStr: string, timestamp: boolean): string => {
+      const date = new Date(dateStr);
+      return date.toLocaleDateString(undefined, {
+        month: "long",
+        day: "numeric",
+        year: "numeric",
+        ...(timestamp ? { hour: "numeric", minute: "numeric" } : {}),
+      });
+    };
+    const actions = (
+      <NotificationActionButton onClick={onActivateWorkspace}>
+        Activate
+      </NotificationActionButton>
+    );
+    notifications.push({
+      actions,
+      title: "Workspace is dormant",
+      severity: "warning",
+      detail: workspace.deleting_at ? (
+        <>
+          This workspace has not been used for{" "}
+          {formatDistanceToNow(Date.parse(workspace.last_used_at))} and was
+          marked dormant on {formatDate(workspace.dormant_at, false)}. It is
+          scheduled to be deleted on {formatDate(workspace.deleting_at, true)}.
+          To keep it you must activate the workspace.
+        </>
+      ) : (
+        <>
+          This workspace has not been used for{" "}
+          {formatDistanceToNow(Date.parse(workspace.last_used_at))} and was
+          marked dormant on {formatDate(workspace.dormant_at, false)}. It is not
+          scheduled for auto-deletion but will become a candidate if
+          auto-deletion is enabled on this template. To keep it you must
+          activate the workspace.
+        </>
+      ),
+    });
+  }
+
+  // Pending in Queue
+  const [showAlertPendingInQueue, setShowAlertPendingInQueue] = useState(false);
+  // 2023-11-15 - MES - This effect will be called every single render because
```

Comment:
Hoping to finally look into this after I get the first Backstage release done this sprint